File: feel-engine-wrapper/src/main/java/io/holunda/feel/rest/ExamineErrorExpressionResource.java
package io.holunda.feel.rest;

import io.holunda.feel.core.ConnectorHelper;
import jakarta.ws.rs.Consumes;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;

@Path("/examineErrorExpression")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
public class ExamineErrorExpressionResource {

    @POST
    public Object examimeErrorExpression(ExamineErrorExpressionRequest request) {
        return ConnectorHelper.examineErrorExpression(request.context(), request.errorExpression()).orElse(null);
    }
}

File: feel-engine-wrapper/src/main/kotlin/io/holunda/feel/rest/ExamineErrorExpressionResource.kt
package io.holunda.feel.rest

import io.holunda.feel.core.*
import jakarta.ws.rs.*
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.MediaType.*

@Path("/examineErrorExpression")
@Consumes(APPLICATION_JSON)
@Produces(APPLICATION_JSON)
class ExamineErrorExpressionResource {

    @POST
    fun examimeErrorExpression(request: ExamineErrorExpressionRequest) =
        FeelEngineWrapper.examineErrorExpression(request.context, request.errorExpression)
}

data class ExamineErrorExpressionRequest(val context: Any, val errorExpression: String)
